src/sentry/sentry_apps/api/bases/sentryapps.py

```diff
@@ -154,7 +157,7 @@ def _get_organization_slug(self, request: Request):
         organization_slug = request.data.get("organization")
         if not organization_slug or not isinstance(organization_slug, str):
             error_message = "Please provide a valid value for the 'organization' field."
-            raise SentryAppError(ResourceDoesNotExist(error_message))
+            raise SentryAppError(message=error_message)
```

Comment:
`ResourceDoesNotExist` has status code 404

```suggestion
            raise SentryAppError(message=error_message, status_code=404)
```